enacts/onset/tests/test_agronomy.py

```diff
@@ -62,7 +62,73 @@ def test_spwba_basic():
     
     assert np.allclose(sm, expected)
     assert np.allclose(drainage, 0)
+
+
+def test_spwba_kc():
+    kc_periods = pd.TimedeltaIndex([0, 45, 47, 45, 45], unit="D")
+    kc_params = xr.DataArray(
+        data=[0.2, 0.4, 1.2, 1.2, 0.6], dims=["kc_periods"], coords=[kc_periods]
+    )
+    p_d = xr.DataArray(
+        pd.DatetimeIndex(data=["2000-05-04"]),
+        dims=["X"],
+        coords={"X": [0]},
+    )
+    sm, drainage, et_crop = agronomy.soil_plant_water_balance(
+        precip_sample(),
+        5,
+        60,
+        10,
+        kc_params=kc_params,
+        planting_date=p_d,
+    )
+    kc_inflex = kc_params.assign_coords(
+        kc_periods=kc_params["kc_periods"].cumsum(dim="kc_periods")
+    )
+    planted_since = precip_sample()["T"] - p_d
+    kc = kc_inflex.interp(
+        kc_periods=planted_since, kwargs={"fill_value": 1}
+    )
```

Comment:
This is not entirely fair: in the function, kc/et_crop are defined through the loop. In this test, they are defined vectorially (because they can in that simple case). So the test find the same results in 2 different ways.

I think a simpler precip_sample will allow what you say until a certain point. After that point, it's likely going to be too complicated even then. And I feel that point is coming soon. Unless we have each of the tests run the basic inputs except for 1. I was thinking rather of cumulating them as I go. Then it might do. I can try regardless. Let me know.

Reply:
I don't have a clear picture in mind of what the rewritten test will look like, but a priori I don't like this one very much, even if the code isn't identical to the code being tested. I have to think too hard to figure out what the test is testing,

Reply:
I don't really see what else to do but a reproducibility test...